feat(managers/custom): introduce `customType` field (#23919)
renovatebot/renovate: docs/usage/configuration-options.md

!!! tip
    Look at our [Regex Manager Presets](https://docs.renovatebot.com/presets-regexManagers/), they may have what you need.

### customType

> `customType` should be a valid custom manager name. Currently we only have one ie. `regex`. In future we will be adding more.

Reviewer: ```suggestion
`customType` should be a valid custom manager name. Currently we only have one i.e. `"regex"`. In future we will be adding more.
```

Example:

```json
{

>   "customType": "regex"

Reviewer 2: Add a `matchStrings` field and put this under `regexManagers` so that it is identifiable such.

}
```

> 

Reviewer: Is this really required? It seems too obvious to me.

Author: Following the existing structure and also better to inform user of the usage, than not I guess.

!!! tip
    Look at our [Regex Manager Presets](https://docs.renovatebot.com/presets-regexManagers/), they may have what you need.

### customType

> `customType` should be a valid custom manager name. Currently we only have one i.e. `"regex"`. In future we will be adding more.

Reviewer 3: ```suggestion
`customType` should be a valid custom manager name. Currently we only have one i.e. `"regex"`.
In future we will be adding more.
```

One sentence per line 😉

!!! tip
    Look at our [Regex Manager Presets](https://docs.renovatebot.com/presets-regexManagers/), they may have what you need.

### customType

> `customType` should be a valid custom manager name. Currently we only have one i.e. `"regex"`.
> In future we will be adding more.
> 

Reviewer 4: ```suggestion
```
